modules/common-extensions/src/main/java/com/liferay/common/extensions/preaction/login/api/LoginPreActionCommonExtensionsImpl.java

```diff
@@ -3,17 +3,40 @@
 import com.liferay.portal.kernel.events.ActionException;
 import com.liferay.portal.kernel.events.LifecycleAction;
 import com.liferay.portal.kernel.events.LifecycleEvent;
+
+import com.liferay.portal.kernel.theme.ThemeDisplay;
 import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.WebKeys;
 import org.osgi.service.component.annotations.Component;
 
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
 @Component(
         immediate = true, property = "key=" + PropsKeys.SERVLET_SERVICE_EVENTS_PRE,
         service = LifecycleAction.class
 )
 public class LoginPreActionCommonExtensionsImpl implements LifecycleAction {
 
     @Override
-    public void processLifecycleEvent(LifecycleEvent lifecycleEvent) throws ActionException {
+    public void processLifecycleEvent(LifecycleEvent lifecycleEvent) {
+
+        HttpServletRequest request = lifecycleEvent.getRequest();
+
+        ThemeDisplay themeDisplay = (ThemeDisplay)request.getAttribute(WebKeys.THEME_DISPLAY);
 
+        if(themeDisplay.isSignedIn()){
+            return;
+        }
+        if (!themeDisplay.getURLCurrent().equals("sample-page")){
+            return;
+        }
+        if(!themeDisplay.isSignedIn()){
+            try{
+                lifecycleEvent.getResponse().sendRedirect("hidden-page");
+            }catch (IOException ioException){
+                throw new RuntimeException(ioException);
+            }
+        }
```

Comment:
I think we can improve the logic here (hence the perfomance too) by reducing the number of `if`conditions.

what about using a single condition like:
`if (!themeDisplay.isSignedIn() && themeDisplay.getURLCurrent().contains("/sample-page")` ?

I would also use `contains` method as I think `equals` might not work for some scenarios, like the URL doesn't match exactly with the String input.